packages/react/src/components/groups/index.tsx

```diff
@@ -4,13 +4,50 @@ import { useSlashID } from "../../main";
 type Groups = string[];
 
 type Props = {
-  belongsTo: (groups: Groups) => boolean;
+  belongsTo: string | ((groups: Groups) => boolean);
```

Comment:
Does it make sense to support `string[]` while we're at it?

Reply:
Just saw the comment you made in the ticket description - I think it wouldn't be surprising to treat the `string[]` option as `AND`. Any more complex logic could be handled using the callback. What do you think?

Reply:
I also have some sample snippets in the docs which elaborate on the XOR/AND thing a bit more:
https://github.com/slashid/ng-evangelion/blob/e8314e059c1f851e822680dbfcdd5e6087786538/web/dev-portal/docs/react-sdk/reference/components/groups.md

Reply:
Oh I only just seen your second comment 😓 Github didn't update or something

I think AND being default for an array is probably okay, do you think AND is more common than OR?

Reply:
I`m a bit confused by XOR... Do you mean regular OR?

Anyway, I think it makes sense to support multiple groups easily. Suggestions:
- Modify this to be `callback | string | (string | string[])[]`. The first array level maps to an OR and the second maps to AND. A few examples:
  - callback works as usual
  - `"group1"`, `"[group1]"` and `"[[group1]]"` are sinonimous and checks if the user belongs to group1
  - `["group1", "group2"]` and`[["group1"], ["group2"]]` are sinonimous and checks if the user belongs to either group1 or group2
  - `[["group1", "group2"]]` checks if the user belongs to both group1 and group2
  - `["group1", ["group2", "group3"]]` checks if the user belongs group1 or (it belongs to both group2 and group3)
  - Stretch goal: while at it, maybe support "!group", so that this becomes a complete [DNF](https://en.wikipedia.org/wiki/Disjunctive_normal_form)-like syntax :joy: 
- Instead, maybe add `belongsToAny` and `belongsToAll` props? 
  Not as flexible, but very self-explanatory.

Reply:
I just noticed this suggestion:

```
<Groups
  belongsTo={all(a, b, c)}
  belongsTo={some(a, b, c)}
>
```

I like this a lot! Extra points if you can mix them: `belongsTo={any(all("a", "b"), "c", not("d")}`

Reply:
@paulo-raca thanks for the comment! I would lean towards the most explicit interface, reducing the implicit behaviour to minimum. I honestly do now know what the primary/most common use cases would be though 🤔

Reply:
Yes I meant OR 🤦

Reply:
If we have to choose a default behaviour I prefer the explicitly named inflexible props, they're much more self documenting and simple to grok

```
<Groups
  belongsTo="a"
  belongsTo={groups => groups.includes("a")}
  belongsToAny={[a, b]}
  belongsToAll={[a, b])
/>
```

We can throw if more than one is set